app/main.py

```diff
@@ -1,2 +1,87 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table: list = [None] * self.capacity
+
+    def calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+                self.hash_table[index] is not None
+                and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    def current_max_size(self, threshold: float = 2 / 3) -> float:
+        return self.capacity * threshold
+
+    def resize(self, capacity_multiplier: int = 2) -> None:
+        previous_hash_table = self.hash_table
+
+        self.__init__(self.capacity * capacity_multiplier)
```

Comment:
Calling `self.__init__` within the `resize` method is not a recommended practice as it can lead to unexpected behavior. Instead, manually update the attributes like `capacity`, `size`, and `hash_table` to reflect the new state after resizing.

Reply:
Could you give me example of unexpected behavior?